patching/plugin-commands-patching/src/patchCommit.ts

```diff
@@ -129,3 +136,42 @@ function removeTrailingAndLeadingSlash (p: string) {
   }
   return p
 }
+
+async function filterIgnoredFiles (folder: string, includedFiles: string[]) {
+  const destinationDir = tempy.directory()
+  await fs.promises.mkdir(destinationDir, { recursive: true })
+
+  const files = await fs.promises.readdir(folder)
+  for (const file of files) {
+    if (includedFiles.includes(file)) {
+      const sourcePath = path.join(folder, file)
+      const destinationPath = path.join(destinationDir, file)
+
+      const stats = await fs.promises.lstat(sourcePath)
+      if (stats.isDirectory()) {
+        await copyDirectory(sourcePath, destinationPath)
+      } else if (stats.isFile()) {
+        await fs.promises.copyFile(sourcePath, destinationPath)
+      }
+    }
+  }
+
+  return destinationDir
+}
+
+async function copyDirectory (source: string, destination: string) {
+  await fs.promises.mkdir(destination, { recursive: true })
```

Comment:
we can reuse the existing  [copyPkg](https://github.com/pnpm/pnpm/blob/main/fs/indexed-pkg-importer/src/index.ts#L185) method.

Reply:
This function seems to copy package.json. But my intention is to copy a certain folder and all the files inside the folder. Do you mean I should refer to this function and write a copyDirectory method and reuse it?